### ISSUE TITLE
ai will move units to attack destructable terrain with units that can not engage

### ISSUE THREAD

**Issue by KBraid** [Robosturm/Commander_Wars#960]
ai will clump infantry and other units that are too weak to attack walls and pipe seams around said structures clearly with the intent to attack but will not do so for obvious reasons

**Comment by KBraid** [comment]
another issue i've found, the ai will try to take the most direct route to attack a structure, even if that route is impassable and there's an open path that it could take.

![image](https://user-images.githubusercontent.com/53922323/163683012-c333f93a-9ee9-4b51-bd69-52bb7ccd5e1a.png)

**Comment by Robosturm** [comment]
Those are just maps the ai can't and won't be able to deal with.
The ai isn't activly moving units to something like the gate but moves units away from production buildings leading to a situation like this.
I did a minor change for ai's to take the longer route if the path is blocked by to many units and the longer path is still relative short, but overall those are situations an ai will fail to deal with in any meaningful way
